pyswarms/backend/handlers.py

```diff
@@ -0,0 +1,163 @@
+            * "intermediate"
+            * "resample"
+            For a description of these see above.
+        """
+        self.strategy = strategy
+
+    def __call__(self, position, bounds, *args, **kwargs):
+        """Make class callable
+
+        The BoundaryHandler can be called as a function to use the strategy
+        that is passed at initialization to repair boundary issues. An example
+        for the usage:
+
+        .. code-block :: python
+
+            from pyswarms.backend import operators as op
+            from pyswarms.backend.handlers import BoundaryHandler
+
+            bh = BoundaryHandler(strategy="reflective")
+            ops.compute_position(swarm, bounds, handler=bh)
+
+
+        Parameters
+        ----------
+        position : np.ndarray
+            The swarm position to be handled
+        bounds : tuple of :code:`np.ndarray` or list
+            a tuple of size 2 where the first entry is the minimum bound while
+            the second entry is the maximum bound. Each array must be of shape
+            :code:`(dimensions,)`
+        *args : tuple
+        **kwargs : dict
+
+        Returns
+        -------
+        numpy.ndarray
+            the adjusted positions of the swarm
+        """
+        # Assign new attributes
+        self.position = position
+        self.lower_bound, self.upper_bound = bounds
+        self.__out_of_bounds()
+
+        if self.strategy == "nearest":
+            new_position = self.nearest()
+        elif self.strategy == "reflective":
+            new_position = self.reflective()
+        elif self.strategy == "shrink":
+            new_position = self.shrink()
+        elif self.strategy == "random":
+            new_position = self.random()
+        elif self.strategy == "intermediate":
+            new_position = self.random()
+        elif self.strategy == "resample":
+            new_position = self.resample()
+
+        return self.position
+
+    def __out_of_bounds(self):
+        """
+        Helper method to find indices
+
+        This helper methods finds the indices of the positions that do
+        transgress the imposed bounds and stores them in class attributes
+        """
+        self.greater_than_bound = np.nonzero(self.position > self.upper_bound)
+        self.lower_than_bound = np.nonzero(self.position < self.lower_bound)
+
+    def nearest(self):
+        """
+        Set position to nearest bound
+
+        This method resets particles that exceed the bounds to the nearest
+        available bound. For every axis on which the coordiantes of the particle
+        surpasses the boundary conditions the coordinate is set to the respective
+        bound that it surpasses.
+        """
+        bool_greater = self.position > self.upper_bound
+        bool_lower = self.position < self.lower_bound
+        self.position = np.where(bool_greater, self.upper_bound, self.position)
+        self.position = np.where(bool_lower, self.lower_bound, self.position)
```

Comment:
I think it's better to just return `new_position` rather than updating `self.position`. Lesser side-effects I guess so that we can test them.

Reply:
Understood, I'll see how I can handle that. :+1:

Reply:
Sure, so instead of passing the result of `np.where` into `self.position` (updating `self.position`), just give it to `new_position` and then return it.